### ISSUE TITLE
Falco error log format forwarded to AWS CloudWatch contains elements outside of JSON string - hard to parse

### ISSUE THREAD

**Issue by dzilbermanvmw** [falcosecurity/charts#466]
**Describe the bug**

Falco logs with 'json_output' and other JSON related settings set to 'true' have "extra" (non JSON)  characters after the "log": element:
**"log": "2023-02-06T04:12:27.514498186Z stdout F** {\"hostname\":\"falco-t64nj\".. 
...
**How to reproduce it**

Following the blog: https://aws.amazon.com/blogs/containers/implementing-runtime-security-in-amazon-eks-using-cncf-falco/, I have deployed the following version of Falco Helm chart falcosecurity/falco: 

with the following settings for JSON formatting (from actual pod):

cat /etc/falco/falco.yaml | grep json_ 
json_include_output_property: true
json_include_tags_property: true
json_output: true

When some run-time errors are injected into application pods, Falco logs  contains non JSON characters whereas per blog the "log" elements is expected to be JSON formatted 

**Expected behaviour**

Falco container logs (obtained from **kubectl logs -f <falco_pod>** ) show errors formatted as:
Error File below /etc opened for writing (user=<NA> user_loginuid=-1 command=touch /etc/34 pid=1277 parent=bash pcmdline=bash file=/etc/34 program=touch gparent=<NA> ggparent=<NA> gggparent=<NA> container_id=f687a1640776 image=docker.io/library/nginx) k8s.ns=default k8s.pod=nginx-test2 container=f687a1640776","priority":"Error","rule":"Write below etc","source":"syscall","tags":["filesystem","mitre_persistence"],"time":"2023-02-07T04:20:41.361668091Z", "output_fields": {"container.id":"f687a1640776","container.image.repository":"docker.io/library/nginx","evt.time":1675743641361668091,"fd.name":"/etc/34","k8s.ns.name":"default","k8s.pod.name":"nginx-test2","proc.aname[2]":null,"proc.aname[3]":null,"proc.aname[4]":null,"proc.cmdline":"touch /etc/34","proc.name":"touch","proc.pcmdline":"bash","proc.pid":1277,"proc.pname":"bash","user.loginuid":-1,"user.name":"<NA>"}}
--------------
Entries in AWS CloudWatch logs, Falco group look like:

**"log": "2023-02-06T04:12:27.514498186Z stdout F** {\"hostname\":\"falco-t64nj\",\"output\":\"04:12:27.505280965: Notice Unexpected connection to K8s API Server from container (command=manager --port=8443 --health-addr=:9090 --prometheus-port=8888 --logtostderr --log-denies=false....}}

Because of this JSON formatting, existing parsing code in related solution repo: https://github.com/aws-samples/aws-securityhub-falco-ecs-eks-integration/blob/main/lambda/falco-security-hub-int.py is failing to parse the "log" element for nested JSON using Python syntax like:

    output = entry["log"]["output"]. 
    severity = map_finding_severity(entry["log"]["priority"])

**Screenshots**

[falco_container_error_bas_json.log](https://github.com/falcosecurity/charts/files/10671602/falco_container_error_bas_json.log)

**Environment**

- Falco version:

NAME    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION
falco   falco           1               2023-02-02 16:01:06.777587 -0800 PST    deployed        falco-2.4.7     0.33.1
------
- System info:

- Cloud provider or hardware configuration: AWS
- OS: 

- Kernel: Debian GNU/Linux

- Installation method: Helm chart **falcosecurity/falco**

AWS EKS Kubernetes
**Additional context**
[values.yaml.txt](https://github.com/falcosecurity/charts/files/10671651/values.yaml.txt)

**Comment by dzilbermanvmw** [comment]
Please advise what Falco configuration (attached values.yaml.txt rename to values.yaml pls) to use to get logs formatted in "pure JSON" format as shown in examples at the bottom of the blog:
https://aws.amazon.com/blogs/containers/implementing-runtime-security-in-amazon-eks-using-cncf-falco/

**Comment by dzilbermanvmw** [comment]
Hello, could someone from Falco team please ack & review the issue? It is pretty much a blocker for my project (as I am trying to parse Falco log entries in AWS CloudWatch assuming that "log" element is well formatted JSON) and this possibly configuration issue is breaking it. thanks

**Comment by Issif** [comment]
Hi,

I'm trying to understand your issue, I'm pretty sure it's not related to Falco itself, but more about Firelens (Fluent bit) configuration.

The first log extract you gave is truncated:
![image](https://user-images.githubusercontent.com/4520100/218194377-0002e774-d0a5-4848-bd70-768b5d6c0c6d.png)
but if I'm just adding the beginning, it's a totally valid json:

```json
{
    "hostname": "falco-t64nj",
    "output": "Error File below /etc opened for writing (user=<NA> user_loginuid=-1 command=touch /etc/34 pid=1277 parent=bash pcmdline=bash file=/etc/34 program=touch gparent=<NA> ggparent=<NA> gggparent=<NA> container_id=f687a1640776 image=docker.io/library/nginx) k8s.ns=default k8s.pod=nginx-test2 container=f687a1640776",
    "priority": "Error",
    "rule": "Write below etc",
    "source": "syscall",
    "tags": [
        "filesystem",
        "mitre_persistence"
    ],
    "time": "2023-02-07T04:20:41.361668091Z",
    "output_fields": {
        "container.id": "f687a1640776",
        "container.image.repository": "docker.io/library/nginx",
        "evt.time": 1675743641361668091,
        "fd.name": "/etc/34",
        "k8s.ns.name": "default",
        "k8s.pod.name": "nginx-test2",
        "proc.aname[2]": null,
        "proc.aname[3]": null,
        "proc.aname[4]": null,
        "proc.cmdline": "touch /etc/34",
        "proc.name": "touch",
        "proc.pcmdline": "bash",
        "proc.pid": 1277,
        "proc.pname": "bash",
        "user.loginuid": -1,
        "user.name": "<NA>"
    }
}
```

And now, if I take the `log` field of the .log you provide:

```json
"log": "2023-02-06T04:12:27.514498186Z stdout F {\"hostname\":\"falco-t64nj\",\"output\":\"04:12:27.505280965: Notice Unexpected connection to K8s API Server from container (command=manager --port=8443 --health-addr=:9090 --prometheus-port=8888 --logtostderr --log-denies=false --emit-admission-events=false --log-level=INFO --exempt-namespace=gatekeeper-system --operation=webhook --enable-external-data=false --enable-generator-resource-expansion=false --log-mutations=false --mutation-annotations=false --disable-cert-rotation=false --max-serving-threads=-1 --tls-min-version=1.3 --metrics-backend=prometheus --operation=mutation-webhook --disable-opa-builtin={http.send} pid=8924 k8s.ns=gatekeeper-system k8s.pod=gatekeeper-controller-manager-6c555b7867-cmd5v container=acf944a8bfa5 image=docker.io/openpolicyagent/gatekeeper:v3.10.0 connection=10.0.10.38:50134->172.20.0.1:443)\",\"priority\":\"Notice\",\"rule\":\"Contact K8S API Server From Container\",\"source\":\"syscall\",\"tags\":[\"container\",\"k8s\",\"mitre_discovery\",\"network\"],\"time\":\"2023-02-06T04:12:27.505280965Z\", \"output_fields\": {\"container.id\":\"acf944a8bfa5\",\"container.image.repository\":\"docker.io/openpolicyagent/gatekeeper\",\"container.image.tag\":\"v3.10.0\",\"evt.time\":1675656747505280965,\"fd.name\":\"10.0.10.38:50134->172.20.0.1:443\",\"k8s.ns.name\":\"gatekeeper-system\",\"k8s.pod.name\":\"gatekeeper-controller-manager-6c555b7867-cmd5v\",\"proc.cmdline\":\"manager --port=8443 --health-addr=:9090 --prometheus-port=8888 --logtostderr --log-denies=false --emit-admission-events=false --log-level=INFO --exempt-namespace=gatekeeper-system --operation=webhook --enable-external-data=false --enable-generator-resource-expansion=false --log-mutations=false --mutation-annotations=false --disable-cert-rotation=false --max-serving-threads=-1 --tls-min-version=1.3 --metrics-backend=prometheus --operation=mutation-webhook --disable-opa-builtin={http.send}\",\"proc.pid\":8924}}"
```

Even if I don't know what is this `stdout F` the following is a perfect JSON, only the double quotes `"` are escaped to store everything in a `string` value. 
If we remove the escapes, it's a valid json too:

```json
{
    "hostname": "falco-t64nj",
    "output": "04:12:27.505280965: Notice Unexpected connection to K8s API Server from container (command=manager --port=8443 --health-addr=:9090 --prometheus-port=8888 --logtostderr --log-denies=false --emit-admission-events=false --log-level=INFO --exempt-namespace=gatekeeper-system --operation=webhook --enable-external-data=false --enable-generator-resource-expansion=false --log-mutations=false --mutation-annotations=false --disable-cert-rotation=false --max-serving-threads=-1 --tls-min-version=1.3 --metrics-backend=prometheus --operation=mutation-webhook --disable-opa-builtin={http.send} pid=8924 k8s.ns=gatekeeper-system k8s.pod=gatekeeper-controller-manager-6c555b7867-cmd5v container=acf944a8bfa5 image=docker.io/openpolicyagent/gatekeeper:v3.10.0 connection=10.0.10.38:50134->172.20.0.1:443)",
    "priority": "Notice",
    "rule": "Contact K8S API Server From Container",
    "source": "syscall",
    "tags": [
        "container",
        "k8s",
        "mitre_discovery",
        "network"
    ],
    "time": "2023-02-06T04:12:27.505280965Z",
    "output_fields": {
        "container.id": "acf944a8bfa5",
        "container.image.repository": "docker.io/openpolicyagent/gatekeeper",
        "container.image.tag": "v3.10.0",
        "evt.time": 1675656747505280965,
        "fd.name": "10.0.10.38:50134->172.20.0.1:443",
        "k8s.ns.name": "gatekeeper-system",
        "k8s.pod.name": "gatekeeper-controller-manager-6c555b7867-cmd5v",
        "proc.cmdline": "manager --port=8443 --health-addr=:9090 --prometheus-port=8888 --logtostderr --log-denies=false --emit-admission-events=false --log-level=INFO --exempt-namespace=gatekeeper-system --operation=webhook --enable-external-data=false --enable-generator-resource-expansion=false --log-mutations=false --mutation-annotations=false --disable-cert-rotation=false --max-serving-threads=-1 --tls-min-version=1.3 --metrics-backend=prometheus --operation=mutation-webhook --disable-opa-builtin={http.send}",
        "proc.pid": 8924
    }
}
```

I added `Cloudwatch Logs` as output for Falcosidekick, so I know the json from Falco is valid and can be stored without any issue, the only thing I see is a transformation from Fluent Bit which makes your .py script not usable as it.
